src/ECharts/Legend.purs

```diff
@@ -0,0 +1,91 @@
+module ECharts.Legend where
+
+import Data.Maybe
+import Data.StrMap 
+import Data.Argonaut.Core
+import Data.Argonaut.Encode
+import Data.Argonaut.Combinators
+
+import ECharts.Color
+import ECharts.Common
+import ECharts.Coords
+import ECharts.Style.Text
+import ECharts.Formatter
+
+data LegendItem = LegendString String |
+  LegendItem {
+    "name" :: Maybe String,
```

Comment:
All records that is subrecords or subsubrecords of Options must be optional :( Because of chart.setOption behaviour. It can be invoked like chart.setOption({tooltip: {itemStyle: {normal: {textStyle: {color: "green"}}}}}) -> and all of other fields will be undefined